### PR TITLE
[Accuracy diff No.128] Fix accuracy diff for paddle.nn.functional.softmax_with_cross_entropy API

### PR DESCRIPTION
**Pull request by NKNaN**
经检查发现是 SoftmaxWithCrossEntropyRule 中的形状处理的问题

回测结果（包括 accuracy_7.txt 中的 case 和 error_config.txt 中的 case）：
image

### CODE REVIEW

**Comment by cangtianhuang** [comment]
@cszdrg 帮忙 review 一下吧～

**Review by cszdrg — APPROVED** [review]
